resqpy/olio/base.py

```diff
@@ -90,22 +94,39 @@ def create_xml(self, title=None, originator=None, add_as_part=False):
             extend this method to complete the XML representation, and then finally ensure the node
             is added as a part to the model.
 
+            if `reuse` is True, a side effect of this method may be to modify the uuid of self;
+            calling code should typically look for such a change and if detected, abandon any
+            further work on building or adding the xml node (as it is already complete)
+
         Args:
             title (string): used as the citation Title text; should usually refer to the well name in a
                 human readable way
             originator (string, optional): the name of the human being who created the deviation survey part;
                 default is to use the login name
             add_as_part (boolean): if True, the newly created xml node is added as a part
                 in the model
+            reuse (boolean, default False): if True, the xml for other parts in the model of the same class
+                is considered for reuse and if suitable the uuid of this object is modified
 
         Returns:
-            node: the newly created root node
+            node: the newly created root node, or reused root node as applicable
         """
 
         assert self.uuid is not None
 
+        if reuse:
```

Comment:
Might it be simpler to have this "reuse" logic in a separate method  e.g. `self.try_reuse`, which would return `True` if a duplicate was found?

That way, the derived classes could do something like this in the overwritten `create_xml`:
```python
if reuse and self.try_reuse(): return self.root
ofn = super().create_xml(add_as_part=False, originator=originator)
```

Alternatively, perhaps the method could return a tuple of (node, was_reused), so the calling code could be:
```python
ofn, existing_found = super().create_xml(add_as_part=False, originator=originator, reuse=reuse)
if existing_found:
   return ofn
```

Reply:
Yes, I had thought of the second alternative. The only thing is that then super().create_xml() will be returning something different from the derived classes create_xml(), which makes me think that we should give it a different name!

But looking at the code, I think your first suggestion is better. I'll go down that route and see how things shape up.